rainier-core/src/main/scala/com/stripe/rainier/sampler/Ehmc.scala

```diff
@@ -0,0 +1,41 @@
+package com.stripe.rainier.sampler
+
+import scala.collection.mutable.ListBuffer
+
+/**
+  * Empirical HMC - automiating tuning of step size and number of leapfrog steps
+  * @param l0 the initial number of leapfrog steps to use during the dual averaging phase
+  * @param k the number of iterations to use when determining the
+  * empirical distribution of the total number of leapfrog steps until a u-turn
```

Comment:
There isn't any guidance in the paper for the value of l0, however it is suggested that a dual averaging scheme could be implemented to determine the initial value l0 for a given target acceptance rate. As for k, 2,000 iterations are used in all the examples in the paper so that's what I've been using.